rcl/src/rcl/wait.c

```diff
@@ -111,6 +112,9 @@ rcl_wait_set_init(
   wait_set->impl->rmw_subscriptions.subscriber_count = 0;
   wait_set->impl->rmw_guard_conditions.guard_conditions = NULL;
   wait_set->impl->rmw_guard_conditions.guard_condition_count = 0;
+  wait_set->impl->rmw_waitset = rmw_create_waitset(
+    NULL, 2*number_of_subscriptions + number_of_guard_conditions);
```

Comment:
You should check to make sure this function succeeded, and if it doesn't you should `goto fail` and make sure it is cleaned up in `__wait_set_clean_up`.